samza-core/src/main/java/org/apache/samza/runtime/AbstractApplicationRunner.java

```diff
@@ -133,4 +139,12 @@ final void writePlanJsonFile(String planJson) {
       log.warn("Failed to write execution plan json to file", e);
     }
   }
+
+  private void validateConfig() {
+    String appClassName = new ApplicationConfig(config).getAppClass();
+    if (appClassName != null && !config.containsKey(TaskConfig.WINDOW_MS())) {
```

Comment:
Why is task.window.ms always required, even if the application doesn't use any window operator?